packages/bff/src/model/api/converters/catalogClientApiConverter.ts

```diff
@@ -0,0 +1,211 @@
+  tenantMailKind,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { attributeNotExists } from "../../domain/errors.js";
+import { agreementApiState } from "../agreementTypes.js";
+import { descriptorApiState } from "../catalogTypes.js";
+
+export function toEserviceCatalogProcessQueryParams(
+  queryParams: bffApi.BffGetCatalogQueryParam
+): catalogApi.GetCatalogQueryParam {
+  return {
+    ...queryParams,
+    eservicesIds: [],
+    name: queryParams.q,
+  };
+}
+
+export function toBffCatalogApiEService(
+  eservice: catalogApi.EService,
+  producerTenant: tenantApi.Tenant,
+  hasCertifiedAttributes: boolean,
+  isRequesterEqProducer: boolean,
+  activeDescriptor?: catalogApi.EServiceDescriptor,
+  agreement?: agreementApi.Agreement
+): bffApi.CatalogEService {
+  const isUpgradable = (agreement: agreementApi.Agreement): boolean => {
+    const eserviceDescriptor = eservice.descriptors.find(
+      (e) => e.id === agreement.descriptorId
+    );
+
+    return (
+      eserviceDescriptor !== undefined &&
+      eservice.descriptors
+        .filter((d) => Number(d.version) > Number(eserviceDescriptor.version))
+        .find(
+          (d) =>
+            (d.state === descriptorApiState.PUBLISHED ||
+              d.state === descriptorApiState.SUSPENDED) &&
+            (agreement.state === agreementApiState.ACTIVE ||
+              agreement.state === agreementApiState.SUSPENDED)
+        ) !== undefined
+    );
+  };
+
+  const partialEnhancedEservice = {
+    id: eservice.id,
+    name: eservice.name,
+    description: eservice.description,
+    producer: {
+      id: eservice.producerId,
+      name: producerTenant.name,
+    },
+    isMine: isRequesterEqProducer,
+    hasCertifiedAttributes,
+  };
+
+  return {
+    ...partialEnhancedEservice,
+    ...(activeDescriptor
+      ? {
+          activeDescriptor: {
+            id: activeDescriptor.id,
+            version: activeDescriptor.version,
+            audience: activeDescriptor.audience,
+            state: activeDescriptor.state,
+          },
+        }
+      : {}),
+    ...(agreement
+      ? {
+          agreement: {
+            id: agreement.id,
+            state: agreement.state,
+            canBeUpgraded: isUpgradable(agreement),
+          },
+        }
+      : {}),
+  };
+}
+
+export function toBffCatalogApiDescriptorAttribute(
+  attributes: attributeRegistryApi.Attribute[],
+  descriptorAttributes: catalogApi.Attribute[]
+): bffApi.DescriptorAttribute[] {
+  return descriptorAttributes.map((attribute) => {
+    const foundAttribute = attributes.find((att) => att.id === attribute.id);
+    if (!foundAttribute) {
+      throw attributeNotExists(unsafeBrandId(attribute.id));
+    }
+
+    return {
+      id: attribute.id,
+      name: foundAttribute.name,
+      description: foundAttribute.description,
+      explicitAttributeVerification: attribute.explicitAttributeVerification,
+    };
+  });
+}
+
+export function toBffCatalogApiDescriptorInterface(
```

Comment:
```suggestion
export function toBffCatalogApiDescriptorDoc(
```
since we use it for documents too let's give it a more generic name